fixes #16592 - keep consistent for headline capitalization

junezhang opened the pull request:
Based on PatternFly Outlines Headline Style Capitalization, http://www.patternfly.org/styles/terminology-and-wording/#capitalization
We suggested to keep consistent in headline capitalization, capitalize the first letter of every word in headline.

As last PR #3841, I fixed the menu case, @ohadlevy suggested me to fix all the labels in one PR.
So I re-created one PR for updating all labels capitalization in Foreman.

Please have a review.
Thanks.

junezhang commented:
@ohadlevy, @bbuckingham, @dLobatog, @tbrisker,  thanks for your comments on my last PR.
May I have your attention for reviewing my new PR, please, thanks!

junezhang commented on app/views/subnets/index.html.erb:
```diff
@@ -7,7 +7,7 @@
       <th class="col-md-3"><%= sort :name, :as => s_("Subnet|Name") %></th>
-      <th class="col-md-2"><%= sort :network, :as => s_("Subnet|Network") %></th>
+      <th class="col-md-2"><%= sort :network, :as => s_("Subnet|Network Address") %></th>
       <th class="col-md-3"><%= _("Domains") %></th>
```
When reviewing the page, it is "Network address",  but in the coding it is "Network". so I added "Address" after Network.
If it is wrong, please correct me. Thanks.

tbrisker replied:
For some reason this was changed to network address by the English translation file.

domcleal replied:
Don't create a second string (`Subnet|Network Address`) without removing the first (`Subnet|Network`), either change the existing translation or remove the original. (Not that attrs should be removed from the generated model attrs file.)

Ditto on every other change to strings that are already extracted via the model attrs file, they're duplicated.

tbrisker replied:
@domcleal I'm not quite sure what you mean, aren't the translation files automatically generated? what is the model attrs file?

domcleal replied:
Non-English translations are in Transifex, English translations are in this repo, and model attrs is locale/model_attributes.rb.

junezhang replied:
@domcleal what you meaning is if want to update the strings, fixing the capitalization problem, I need to update the English translation file: locale/en/foreman.po, Is that right?

junezhang replied:
@domcleal Update the English translation PO file, and re-submit a patch, please help to review. Thanks.

tbrisker commented:
@theforeman-bot ok to test

tbrisker commented:
Sorry for the slow response, I was on a long vacation. OK to merge if all tests pass.

junezhang commented:
@domcleal I've updated the English translation PO file. Please help to review? 
When updating the PO file, I haven't found how to update these two items (Provisioning templates, Smart class parameters) in the navigation, so update them in the menu/loader.rb file.
Thanks.

ohadlevy commented:
@junezhang can you please rebase?
@tbrisker anything else this PR needs? thanks!

junezhang committed:
fixes #16592 - keep consistent for headline capitalization

Based on PatternFly Outline Headline Style, Suggested to
keep consistent in headline capitaliztion.

junezhang commented:
test

junezhang commented:
Rebase my commit, @tbrisker @domcleal Please have a review. Thanks.

junezhang commented:
It's a long time for seeking feedback about my English translation PO file. so I think maybe it's not a right way. then I turn back to update the menu/loader.rb file and other .erb files. Please help to review it. Thanks.

sean797 commented:
Thanks a ton for doing this @junezhang 🥇 Looks good overall. If you have a chance to look at the test failures, can you? When they pass, it should be good to merge. If not I will raise a PR based off this one.

junezhang commented:
Hi, @sean797, sorry for replying late. I've checked the test result, but have no idea about the test failure. Could you help on the test error, please? Thank you!

sean797 commented:
Sure, try this @junezhang 

```
$ git diff
diff --git a/test/integration_test_helper.rb b/test/integration_test_helper.rb
index 0c0e109..e1f7b3a 100644
--- a/test/integration_test_helper.rb
+++ b/test/integration_test_helper.rb
@@ -37,14 +37,14 @@ class ActionDispatch::IntegrationTest
   def assert_index_page(index_path,title_text,new_link_text = nil,has_search = true,has_pagination = true)
     visit index_path
     assert page.has_selector?('h1', :text => title_text), "#{title_text} was expected in the <h1> tag, but was not found"
-    (assert find_link(new_link_text).visible?, "#{new_link_text} is not visible") if new_link_text
+    (assert first(:link, new_link_text).visible?, "#{new_link_text} is not visible") if new_link_text
     (assert find_button('Search').visible?, "Search button is not visible") if has_search
     (assert has_content?("Displaying"), "Pagination 'Display ...' does not appear") if has_pagination
   end
 
   def assert_new_button(index_path,new_link_text,new_path)
     visit index_path
-    click_link new_link_text
+    first(:link, new_link_text).click
     assert_current_path new_path
   end

dLobatog reviewed (changes requested):
Set to Waiting on contributor due to the failing test, thanks for the help @junezhang @sean797

junezhang commented:
@sean797  Thanks for providing suggestion to fix the test failure. Currently I'm traveling in US. I will fix it next week. Sorry for any inconvenience.

junezhang commented:
@sean797 I've submitted a commit, but it is still failed. So could you help to raise a PR based off this one, please?

sean797 commented:
@junezhang it works when I run them locally. Lets [test foreman] again, looks like it could be an intermittent failure

sean797 reviewed (approved):
\o/ tests passed, thanks @junezhang 👍

junezhang commented:
@sean797 Thanks for your supporting. 
@dLobatog please have a review for the capitalization updating.

ohadlevy commented:
@dLobatog bump?

dLobatog commented:
Thank you @junezhang & @sean797

dLobatog commented on app/views/media/index.html.erb:
```diff
@@ -9,7 +9,7 @@
       <th class="col-md-2"><%= sort :name, :as => s_("Medium|Name") %></th>
       <th class="col-md-6"><%= sort :path, :as => s_("Medium|Path") %></th>
-      <th class="col-md-1"><%= sort :family, :as => s_("Medium|Os family") %></th>
+      <th class="col-md-1"><%= sort :family, :as => s_("Medium|Os Family") %></th>
```
I think I may have missed this, should be `OS Family`.

dLobatog commented on app/views/ptables/index.html.erb:
```diff
@@ -7,8 +7,8 @@
       <th class="col-md-4"><%= sort :name, :as => s_("Ptable|Name") %></th>
-      <th class="col-md-1"><%= sort :family, :as => s_("Ptable|Os family") %></th>
-      <th class="col-md-4"><%= _("Operating systems") %></th>
+      <th class="col-md-1"><%= sort :family, :as => s_("Ptable|Os Family") %></th>
```
Another one, OS family.

sean797 commented:
ah nice spot @dLobatog ! Created https://github.com/theforeman/foreman/pull/4700

ares commented:
I set 1.16 in redmine issue and marked the linked PR to be included in 1.16 too.
